src/Block.ts

```diff
@@ -0,0 +1,112 @@
+import { BytesLike, SignatureLike, Signer, ethers } from "ethers";
+import { BlockStruct } from "@typechain-types/contracts/V1/types/DataTypes";
+import { Codec, Type } from "./utils/Codec";
+import {
+    ForkId,
+    BlockHeight,
+    BlockTimestamp,
+    Address,
+    ChannelId,
+    Hash,
+    Signature
+} from "@/types/types";
+import { SignedBlockStruct } from "@typechain-types/contracts/V1/types/DataTypes";
+
+export class Block {
+    private constructor(private readonly block: BlockStruct) {}
+
+    static from(block: BlockStruct): Block {
+        return new Block(block);
```

Comment:
Any reason/benefit of having this and not just doing `new Block(blockStruct))`

Reply:
Nicer syntax. Flows better.